hudi-common/src/main/java/org/apache/hudi/common/table/read/BufferedRecords.java

```diff
@@ -48,6 +48,24 @@ public static <T> BufferedRecord<T> fromHoodieRecord(HoodieRecord record, Schema
     return new BufferedRecord<>(recordKey, recordContext.convertOrderingValueToEngineType(orderingValue), data, schemaId, inferOperation(isDelete, record.getOperation()));
   }
 
+  public static <T> BufferedRecord<T> fromHoodieRecordWithDeflatedRecord(HoodieRecord record, Schema schema, RecordContext<T> recordContext, Properties props,
+                                                                         String[] orderingFields, DeleteContext deleteContext) {
+    HoodieOperation hoodieOperation = record.getIgnoreIndexUpdate() ? HoodieOperation.UPDATE_BEFORE : record.getOperation();
```

Comment:
This needs to use `inferOperation` to ensure consistency with the current path. That may explain the flink failure.

Reply:
down the line, we are using inferOperation. will check it out anyways on whats going on.